contracts/token/common/ERC2981/ERC2981Internal.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.8;
+
+import { ERC2981Storage } from './ERC2981Storage.sol';
+import { IERC2981Internal } from './IERC2981Internal.sol';
+
+/**
+ * @title ERC2981 internal functions
+ */
+abstract contract ERC2981Internal is IERC2981Internal {
+    using ERC2981Storage for ERC2981Storage.Layout;
+
+    function _royaltyInfo(uint256 tokenId, uint256 salePrice)
+        internal
+        view
+        virtual
+        returns (address, uint256)
+    {
+        ERC2981Storage.Layout storage l = ERC2981Storage.layout();
+
+        uint256 royalty = l.royalties[tokenId] > 0
+            ? l.royalties[tokenId]
+            : l.royalty;
```

Comment:
Move this to an internal `_getRoyaltyBPS(uint256 tokenId)` function.

Reply:
Makes it easier for the user to override this logic.

Reply:
What does BPS stand for? 

Also, does it make sense to add the `Layout` as a parameter? It reduces the number of times the storage layout is accessed.

```js
function _getRoyaltyBPS(ERC2981Storage.Layout storage l, uint256 tokenId)
```

Reply:
https://www.investopedia.com/terms/b/basispoint.asp